File: api/src/main/java/fr/huiitre/tools/modules/elite_dangerous/r2r/infrastructure/importer/SpanshJsonRouteImporter.java
package fr.huiitre.tools.modules.elite_dangerous.r2r.infrastructure.importer;

import java.io.IOException;
import java.nio.charset.StandardCharsets;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

import fr.huiitre.tools.modules.elite_dangerous.r2r.application.importer.RouteImporter;

public class SpanshJsonRouteImporter implements RouteImporter {

    private static final ObjectMapper MAPPER = new ObjectMapper();

    @Override
    public String source() {
        return "spansh";
    }

    @Override
    public String parse(byte[] fileContent, String fileName) throws IOException {
        String json = new String(fileContent, StandardCharsets.UTF_8);
        JsonNode root = MAPPER.readTree(json);
        if (!root.has("result") || !root.get("result").isArray()) {
            throw new IllegalArgumentException("R2R_INVALID_JSON_FORMAT");
        }
        return json;
    }
}

File: api/Modules/EliteDangerous/RoadToRiches/Infrastructure/SpanshJsonRouteImporter.cs
using System.Text;
using System.Text.Json;
using Tools.Api.Modules.Core.Common.Application.Exceptions;
using Tools.Api.Modules.EliteDangerous.RoadToRiches.Application.Ports;

namespace Tools.Api.Modules.EliteDangerous.RoadToRiches.Infrastructure;

public sealed class SpanshJsonRouteImporter : IRouteImporter
{
    public string Source => "spansh";

    // Le document entier est conservé : le frontend lit `parameters` autant que `result`.
    public string Parse(byte[] fileContent, string fileName)
    {
        using var document = JsonDocument.Parse(fileContent);

        if (!document.RootElement.TryGetProperty("result", out var result)
            || result.ValueKind != JsonValueKind.Array)
        {
            throw AppException.Validation(
                "R2R_INVALID_JSON_FORMAT",
                $"Le fichier {fileName} n'est pas un export Spansh valide.");
        }

        return Encoding.UTF8.GetString(fileContent);
    }
}
